### PR TITLE
Added documentation for modes

### PR DESCRIPTION
**Pull request by dherrada**
@tannewt This should clear up #16. I tried to document them where they were defined, but sphinx was not cooperating

### CODE REVIEW

File: adafruit_bno055.py [hunk 1]
```diff
@@ -143,44 +143,110 @@ def _reset(self):
...
+            | NDOF_MODE        |   X   |   X     |  X   |     X    |
+            +------------------+-------+---------+------+----------+
 
         The default mode is ``NDOF_MODE``.
+
+        | You can set the mode using the line below:
+        | ``sensor.mode = adafruit_bno055.ACCONLY_MODE``
+        | replacing ``ACCONLY_MODE`` with the mode you want to use
+
+        .. data:: CONFIG_MODE
```
Comment (tannewt) [inline adafruit_bno055.py:185]: Does this render it on the module or class?

Comment (dherrada) [reply adafruit_bno055.py:185]: Class. Here's what it looks like. I can have it render on the module, so it starts right after the author, but I thought this way was a bit cleaner.

docs

**Comment by dherrada** [comment]
@tannewt So I wasn't able to get it to render that way. I'm not sure why, but I'm not very experienced with autodoc or sphinx, so there's probably a way to do it that I just don't know about. 

I tried putting comments with `#:` on the same line and putting docstrings with just text and with `.. data::` after the lines and none of that worked.

**Comment by siddacious** [comment]
@dherrada I use this tool to make tables for RST docs:
https://www.tablesgenerator.com/text_tables

I'll look for an example of when I used it.. 👀

**Comment by siddacious** [comment]
@dherrada The HTS221 docs use it:
https://circuitpython.readthedocs.io/projects/hts221/en/latest/api.html
from 
https://github.com/adafruit/Adafruit_CircuitPython_HTS221/blob/master/adafruit_hts221.py#L102

**Comment by dherrada** [comment]
@siddacious I think there may be an example [in this library already](https://github.com/adafruit/Adafruit_CircuitPython_BNO055/blob/master/adafruit_bno055.py#L152-L181). The issue I'm currently running into is that it'd be better to document the modes where they are defined, but I can't figure out how to tell sphinx to do that.

Also, FWIW, the tables render a lot better in rtd if you put them in a code block because they rely on a monospaced font to look right

**Comment by dherrada** [comment]
Oh... Huh. I wonder why it didn't work like that here

**Comment by tannewt** [comment]
> I tried putting comments with `#:` on the same line and putting docstrings with just text and with `.. data::` after the lines and none of that worked.

So if you did a triple quoted comment after the definition it didn't work? That may be a class member thing only.

**Comment by dherrada** [comment]
Yeah, I think that may be the case

**Comment by dherrada** [comment]
@tannewt Moving forward, what would you like me to do with this PR? I know the way I have it isn't really ideal, but I'm not quite sure how else to do it.

**Pull request merged**